clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/model/CloudFoundryLoadBalancer.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.cloudfoundry.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.netflix.spinnaker.clouddriver.cloudfoundry.CloudFoundryCloudProvider;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancer;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancerInstance;
+import com.netflix.spinnaker.clouddriver.model.LoadBalancerServerGroup;
+import com.netflix.spinnaker.moniker.Moniker;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+import static java.util.stream.Collectors.toSet;
+
+@RequiredArgsConstructor
+@ToString
+@EqualsAndHashCode(of = "id", callSuper = false)
+@Getter
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CloudFoundryLoadBalancer extends CloudFoundryModel implements LoadBalancer, Cloneable {
+  private static final Moniker EMPTY_MONIKER = new Moniker();
+
+  private final String account;
+  private final String id;
+  @Nullable
+  private final String host;
+  @Nullable
+  private final String path;
+  @Nullable
+  private final Integer port;
+  private final CloudFoundrySpace space;
+  private final CloudFoundryDomain domain;
+
+  @JsonIgnore
+  private final Set<CloudFoundryServerGroup> mappedApps;
+
+  @JsonProperty
+  public String getName() {
+    return host + "." + domain + "." + getName() + (port == null ? "" : "-" + port) + (path == null ? "" : "/" + path);
+  }
+
+  @Override
+  public Moniker getMoniker() {
+    return EMPTY_MONIKER;
```

Comment:
I would recommend setting this correctly unless there is an explicit need for keeping this empty (or you plan to implement this later).

Reply:
Yes, there is an explicit need to keep it empty. Cloud Foundry load balancers are inseparable from public routes in a way that a moniker doesn't describe well.

Reply:
In other words, it isn't possible to separate a CF load balancer into cluster, stack, detail, etc.

Reply:
What about the 'app' field?

Reply:
There is also no way to infer an app from a CF load balancer.